benchmark/Streamly/Benchmark/Data/Parser/ParserD.hs

```diff
@@ -165,6 +165,39 @@ longestAllAny value =
             (drainWhile (<= value))
         )
 
+-------------------------------------------------------------------------------
+-- Derived parsers
+-------------------------------------------------------------------------------
+
+{-# INLINE span #-}
+span :: MonadThrow m => Int -> SerialT m Int -> m ()
+span value = IP.parseD (span_ (<= (value `div` 2)) FL.drain FL.drain)
+
+    where
+
+    span_ p f1 f2 =
+        PR.splitWith (\_ _ -> ()) (PR.takeWhile p f1) (PR.fromFold f2)
```

Comment:
The code would be clearer if we define the two parsers p1 and p2 and then use `splitWith f p1 p2`.